File: TypeScript/Algorithm/Algorithm I/344. Reverse String.ts
/**
 * Reverses an array of characters in-place.
 * @param {string[]} s - The array of characters to be reversed.
 * @returns {void}
 */
function reverseString(s: string[]): void {
  const LENGTH: number = s.length;

  for (let index: number = 0; index < LENGTH / 2; index++) {
    const tempChar: string = s[index];

    s[index] = s[LENGTH - index - 1];
    s[LENGTH - index - 1] = tempChar;
  }
}

File: Java/Algorithm/Algorithm I/344. Reverse String.java
class Solution {

  /**
   * Reverses a character array in place.
   *
   * @param s The character array to be reversed.
   */
  public void reverseString(char[] s) {
    int STRING_LENGTH = s.length;

    int startPointer = 0;
    int endPointer = STRING_LENGTH - 1;

    while (startPointer < endPointer) {
      char tempChar = s[startPointer];

      s[startPointer] = s[endPointer];
      s[endPointer] = tempChar;

      startPointer++;
      endPointer--;
    }
  }
}
